tests/integ_tests/workflow_tests/signals.rs

```diff
@@ -0,0 +1,105 @@
+use futures::StreamExt;
+use temporal_sdk_core::prototype_rust_sdk::{ChildWorkflowOptions, WfContext, WorkflowResult};
+use test_utils::CoreWfStarter;
+use uuid::Uuid;
+
+const SIGNAME: &str = "signame";
+const RECEIVER_WFID: &str = "sends-signal-signal-receiver";
+
+async fn signal_sender(mut ctx: WfContext) -> WorkflowResult<()> {
+    let run_id = std::str::from_utf8(&ctx.get_args()[0].data)
+        .unwrap()
+        .to_owned();
+    let sigres = ctx
+        .signal_workflow(RECEIVER_WFID, run_id, SIGNAME, b"hi!")
+        .await;
+    if ctx.get_args().get(1).is_some() {
+        // We expect failure
+        assert!(sigres.is_err());
+    } else {
+        sigres.unwrap();
+    }
+    Ok(().into())
+}
+
+#[tokio::test]
+async fn sends_signal_to_missing_wf() {
+    let wf_name = "sends_signal_to_missing_wf";
+    let mut starter = CoreWfStarter::new(wf_name);
+    let worker = starter.worker().await;
+    worker.register_wf(wf_name.to_owned(), signal_sender);
+
+    worker
+        .submit_wf(
+            wf_name,
+            wf_name,
+            vec![Uuid::new_v4().to_string().into(), [1].into()],
+        )
+        .await
+        .unwrap();
+    worker.run_until_done().await.unwrap();
+    starter.shutdown().await;
+}
+
+async fn signal_receiver(mut ctx: WfContext) -> WorkflowResult<()> {
+    ctx.make_signal_channel(SIGNAME).next().await.unwrap();
+    Ok(().into())
+}
+
+#[tokio::test]
+async fn sends_signal_to_other_wf() {
+    let mut starter = CoreWfStarter::new("sends_signal_to_other_wf");
+    let worker = starter.worker().await;
+    worker.register_wf("sender", signal_sender);
+    worker.register_wf("receiver", signal_receiver);
+
+    let receiver_run_id = worker
+        .submit_wf(RECEIVER_WFID, "receiver", vec![])
```

Comment:
```suggestion
        .submit_wf(RECEIVER_WFID, "receiver", EXPECT_SUCCESS_ARGS)
```

Reply:
Your two comments are actually for the two different types of workflows -- just the presence of a second argument in the sender indicates failure. Lifting these out just feels more awkward than not to me

Reply:
It helps with readability.